pkg/virtual/apiexport/controllers/apireconciler/apiexport_apireconciler_controller.go

```diff
@@ -121,17 +136,110 @@ type APIReconciler struct {
 	apiSets map[dynamiccontext.APIDomainKey]apidefinition.APIDefinitionSet
 }
 
+func (c *APIReconciler) gatherInternalAPIs() ([]internalapis.InternalAPI, error) {
+	_, apiResourcesLists, err := c.discoveryClient.ServerGroupsAndResources()
+	if err != nil {
+		return nil, err
+	}
+	internalAPIs := []internalapis.InternalAPI{}
+
+	for _, apiResourcesList := range apiResourcesLists {
+		gv, err := schema.ParseGroupVersion(apiResourcesList.GroupVersion)
+		if err != nil {
+			klog.Errorf("error parsing group version %v, err: %v", apiResourcesList.GroupVersion, err)
+			continue
+		}
+		// ignore kcp resources
+		if strings.HasSuffix(gv.Group, ".kcp.dev") {
+			continue
+		}
+		for _, apiResource := range apiResourcesList.APIResources {
+			// ignore subresources
+			if strings.Contains(apiResource.Name, "/") {
+				continue
+			}
+			resourceScope := apiextensionsv1.ClusterScoped
+			if apiResource.Namespaced {
+				resourceScope = apiextensionsv1.NamespaceScoped
+			}
+
+			gvk := schema.GroupVersionKind{Kind: apiResource.Kind, Version: gv.Version, Group: gv.Group}
+
+			instance, err := genericcontrolplanescheme.Scheme.New(gvk)
+			if err != nil {
+				if extensionsapiserver.Scheme.Recognizes(gvk) {
+					// Not currently supporting permission claim for CustomResourceDefinition.
+					// extensionsapiserver.Scheme is recursive and fails in internalapis.CreateAPIResourceSchemas in schema converter
+					klog.Warningf("permission claims not currently supported for gvk: %v/%v %v", gv.Group, gv.Version, apiResource.Kind)
+				} else {
+					klog.Errorf("error creating instance for gvk: %v/%v %v err: %v", gv.Group, gv.Version, apiResource.Kind, err)
+				}
+				continue
+			}
+
+			raw, err := runtime.DefaultUnstructuredConverter.ToUnstructured(instance)
+			if err != nil {
+				klog.Errorf("error checking status for gvk: %v/%v %v err: %v", gv.Group, gv.Version, apiResource.Kind, err)
+				continue
+			}
+			u := unstructured.Unstructured{Object: raw}
+			_, hasStatus, _ := unstructured.NestedStringMap(u.Object, "status")
```

Comment:
I don't know, but just asking, would status be set here if it is omitempty?

Reply:
Doh! Great catch. I see LocalSubjectAccessReview is coming out false for this reason. Do you have any other thoughts on how to figure this out?

Reply:
@sttts do you have any suggestions here on another way to do this check?

Reply:
Just submitted another comment w/a possible solution

Reply:
https://github.com/kcp-dev/kcp/pull/1676#discussion_r938072016

Reply:
Thank you! I'll get going on this.